src/snowflake/connector/cursor.py

```diff
@@ -1724,13 +1724,38 @@ def get_result_batches(self) -> list[ResultBatch] | None:
 
 class DictCursor(SnowflakeCursor):
     """Cursor returning results in a dictionary."""
-
     def __init__(self, connection) -> None:
         super().__init__(
             connection,
             use_dict_result=True,
         )
 
+    def fetchone(self) -> dict | None:
+        return super().fetchone()
+    
+    def fetchmany(self, size: int | None = None) -> list[dict]:
+        return super().fetchmany()
+    
+    def fetchall(self) -> list[dict]:
+        return super().fetchall()
+
+class TupleCursor(SnowflakeCursor):
```

Comment:
do we need `TupleCursor`? this is effectively the default SnowflakeCursor.

Reply:
`SnowflakeCursor `has to return `Union[None|dict|tuple] `because could be instanciated as
`SnowflakeCursor(use_dict_result=True)`.

`TupleCursor `is just a `SnowflakeCursor `that return `Union[None|tuple]` on fetch

On `DictCursor `the better types will not require nothing for the user, on `TupleCursor `it will require that they start to instanciate as `TupleDict`, so maybe its not useful. I let you to decide, let me know if you want and i delete the `TupleCursor `from PR and issue.